src/components/Select/MultiSelect.tsx

```diff
@@ -0,0 +1,89 @@
+import { useCallback, useEffect, useState } from "react";
+
+import { SelectContainerProps, SelectOptionProp } from "./common/types";
+import { SelectGroup, SelectItem, InternalSelect } from "./common/InternalSelect";
+
+export interface MultiSelectProps
+  extends Omit<
+    SelectContainerProps,
+    "onChange" | "value" | "open" | "onOpenChange" | "onSelect"
+  > {
+  defaultValue?: Array<string>;
+  onChange?: (value: Array<string>) => void;
+  value?: Array<string>;
+  defaultOpen?: boolean;
+}
+
+export const MultiSelect = ({
+  value: valueProp,
+  defaultValue,
+  onChange: onChangeProp,
+  options,
+  children,
+  ...props
+}: MultiSelectProps) => {
+  const [selectedValues, setSelectedValues] = useState<Array<string>>(
+    valueProp ?? defaultValue ?? []
+  );
+  const [open, setOpen] = useState(false);
+  const onOpenChange = useCallback(
+    (newOpen?: boolean) => {
+      setOpen(newOpen ?? !open);
+    },
+    [open]
+  );
+
+  useEffect(() => {
+    setSelectedValues(valueProp ?? []);
+  }, [valueProp]);
+
+  const onChange = useCallback(
+    (values: Array<string>) => {
+      setSelectedValues(values);
+      if (typeof onChangeProp === "function") {
+        onChangeProp(values);
+      }
+    },
+    [onChangeProp]
+  );
```

Comment:
On Change Passes the entire list 
In case on onSelectpeople expect the last clicked element in case of addition of removal
I can add the onSelect if needed

Reply:
i think the onSelect should pass all the selected values so that the user doesn't have to keep track of the changes and we can get rid of the `onChanges` prop

Reply:
Got it will rename onChange with onSelect